<a href="https://colab.research.google.com/github/ju4n1t0x/agroindustria/blob/main/CDD_Grupo24_TPIntegrador.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Trabajo Práctico Integrador - Ciencia de Datos

**Nombre y Apellido:** Stefania Martos, Juan Ignacio Sasia, Juan José Muñoz Franchi, Jorge Nicolas Segovia. <br>
**Materia:** Ciencia de Datos  <br>
**Docente:** Germán Giró, Malena Constancio, Giselle Bustamante<br>
**Fecha:** 26/04/2026  <br>
**Unidad:** Módulo 1 y 2  <br>
**Número de ejercicio/actividad:** Entrega 1  

## 1. Objetivo

El objetivo de este trabajo es analizar un conjunto de datos reales sobre movimientos de ganado bovino en Argentina, con la intención de entender cómo funciona este sistema y qué tipo de patrones pueden aparecer.

En esta primera entrega, el foco está en definir bien el problema, entender el contexto del dominio y hacer una primera exploración del dataset para ver qué tipo de información contiene y en qué estado se encuentra.

## 2. Dataset

Fuente:  
https://datos.gob.ar/dataset/agroindustria-senasa---movimientos-bovinos/archivo/agroindustria_81330b51-8dd9-420d-842e-aefe56151d3a

##3.1.A - Comprensión del negocio


###Contexto / dominio del problema

El presente trabajo pertenece al area de la produccion y logistica ganadera bovina en Argentina, un punto clave de la agroindustria nacional.

El dataset proviene de SENASA(Servicio Nacional de Sanidad y Calidad Agroalimentaria) y contiene informacion sobre movimientos mensuales de ganado bovino entre distintas regiones del pais, desagregador por categoria animal (vacas, novillos, terneros, etc).

Estos movimientos corresponden a distintas etapas del ciclo productivo como: **cria, recria, engorde (por ejemplo en feedlots) y traslado a faena.**

El seguimiento de estos movimientos es fundamental para:

- el control sanitario  
- la organización logistica del sector  
- la sostenibilidad de la cadena productiva  
- entre otros  

Diversos actores pueden beneficiarse de este analisis:

- organismos publicos (control y regulacion)  
- productores ganaderos (toma de decisiones)  
- frigorificos (planificacion de abastecimiento)  
- investigadores (modelado de redes y riesgos sanitarios)  


###Problema o pregunta que se busca responder con los datos:

El sistema de movimientos bovinos en Argentina es parte de una red compleja que involucra diferentes regiones y etapas productivas.

Sin embargo, no se entiende claramente cómo está organizada esta red ni cómo evoluciona en el tiempo, lo que dificulta tanto su análisis como la toma de decisiones en situaciones críticas.

Esta falta de comprensión impacta la sanidad animal, la eficiencia logística y el funcionamiento del sistema productivo y comercial del sector agrónomo.



###Objetivo general

El objetivo general de nuestro trabajo es analizar la estructura y la dinámica de los movimientos bovinos en Argentina para identificar patrones relevantes, que contribuyan a comprender el sistema logístico y productivo del sector ganadero.


###Objetivos específicos

- Identificar las regiones con mayor participación en los flujos de ganado, destacando posibles nodos relevantes dentro de la red.  
- Analizar la evolución temporal de los movimientos para detectar posibles patrones estacionales o cambios en el comportamiento del sistema.  
- Evaluar las diferencias en los patrones de movimiento según la categoría animal, como aproximación a las distintas etapas del ciclo productivo.  
- Identificar concentraciones o dependencias dentro de la red que puedan representar puntos críticos del sistema.  


###Preguntas analíticas

Las siguientes preguntas guiarán el análisis en las etapas posteriores:

- ¿Qué regiones presentan mayor volumen de entrada y salida de ganado bovino, y cómo se distribuyen estos flujos a nivel nacional?  
- ¿Existen patrones temporales o estacionales en los movimientos bovinos a lo largo del tiempo?  
- ¿Cómo varían los patrones de movimiento según la categoría animal (por ejemplo, terneros vs. novillos), y qué implicancias tiene esto en términos del ciclo productivo?  
- ¿Se pueden identificar regiones que actúan como nodos centrales dentro de la red de movimientos bovinos?  

##3.1.B - Comprensión de los datos

In [ ]:
from csv import reader, writer
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline

###***Fuente de los datos***



El dataset fue obtenido del portal de datos abiertos de Argentina (datos.gob.ar), y es publicado por SENASA.

Contiene información sobre los movimientos de ganado bovino dentro del país, incluyendo datos de origen, destino, fecha y cantidad de animales según su categoría.

Los datos están disponibles en formato CSV y para esta entrega se utilizó el archivo correspondiente al año 2018.

###***Carga de los datos***



A continuación se realiza la carga del dataset utilizando la librería Pandas para su posterior análisis.

In [ ]:
# Cargamos el dataset
df = pd.read_csv("movimiento-bovinos-2018.csv", encoding="latin1")

# Convertimos la columna 'fecha' a formato de fecha (datetime)
df['fecha'] = pd.to_datetime(df['fecha'])

# Mostramos las primeras filas del dataset para ver cómo vienen los datos
df.head(10)

,fecha,salido_de,ingresado_a,provincia_origen,provincia_origen_id,departamento_origen,departamento_origen_id,provincia_destino,provincia_destino_id,departamento_destino,departamento_destino_id,vaca,vaquillona,novillo,novillito,ternero,ternera,torito,toro,bueyes
0,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,25 de Mayo,6854,212,150,0,342,931,566,0,21,0
1,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,9 de Julio,6588,35,0,0,0,96,104,0,4,0
2,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Alberti,6021,0,0,0,0,31,9,0,0,0
3,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Bolívar,6105,0,0,0,0,29,39,0,0,0
4,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Bragado,6112,0,0,0,0,11,14,0,0,0
5,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Chivilcoy,6224,0,0,0,0,35,30,0,0,0
6,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,General Alvear,6287,0,0,0,0,0,0,0,5,0
7,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,General Viamonte,6385,0,0,0,0,22,28,0,30,0
8,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Hipólito Yrigoyen,6406,0,0,0,0,140,0,0,0,0
9,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Lobos,6483,50,7,0,6,10,10,0,0,0


**Visualizamos que el dataset con el que vamos a trabajar posee el siguiente orden de columnas:**
- `fecha`
- `salido_de`
- `ingresado_a`
- `provincia_origen`
- `provincia_origen_id`
- `departamento_origen`
- `departamento_origen_id`
- `provincia_destino`
- `provincia_destino_id`
- `departamento_destino`
- `departamento_destino_id`
- `vaca`
- `vaquillona`
- `novillo`
- `novillito`
- `ternero`
- `ternera`
- `torito`
- `toro`
- `bueyes`

In [ ]:
# Vemos la cantidad de filas y columnas del dataset
df.shape

(168802, 20)

**El dataset posee un total de 168802 filas y 20 columnas**

In [ ]:
# Información general: tipos de datos, cantidad de valores y nulos
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 168802 entries, 0 to 168801
Data columns (total 20 columns):
 #   Column                   Non-Null Count   Dtype         
---  ------                   --------------   -----         
 0   fecha                    168802 non-null  datetime64[ns]
 1   salido_de                168802 non-null  object        
 2   ingresado_a              168802 non-null  object        
 3   provincia_origen         168802 non-null  object        
 4   provincia_origen_id      168802 non-null  int64         
 5   departamento_origen      168802 non-null  object        
 6   departamento_origen_id   168802 non-null  int64         
 7   provincia_destino        168802 non-null  object        
 8   provincia_destino_id     168802 non-null  int64         
 9   departamento_destino     168802 non-null  object        
 10  departamento_destino_id  168802 non-null  int64         
 11  vaca                     168802 non-null  int64         
 12  vaquillona      

###**Exploramos los tipos de datos de las 20 para luego armar el diccionario**

In [ ]:
# Contamos la cantidad de valores únicos en cada columna
df.nunique()

,0
fecha,12
salido_de,3
ingresado_a,3
provincia_origen,24
provincia_origen_id,24
departamento_origen,393
departamento_origen_id,462
provincia_destino,24
provincia_destino_id,24
departamento_destino,398


**Verificamos a partir de esto que contamos con:**
- medicion correspondiente a 12 meses
- que hay 3 tipos de establecimientos de salida y de origen
- que los registros comprenden las 24 provincias pero un total de 393 departamentos de origen y 471 departamentos de destino


In [ ]:
departamentos_descripcion_unicos = df.groupby('departamento_destino')['departamento_destino_id'].nunique()
departamentos_descripcion_unicos

,departamento_destino_id
departamento_destino,
12 de Octubre,1
1° de Mayo,1
2 de Abril,1
25 de Mayo,5
9 de Julio,4
...,...
Yerba Buena,1
Zapala,1
Zárate,1


Podemos ver que hay departamentos que poseen la misma descripcion pero lo podemos encontrar con numeros de id diferentes, lo que teniendo en cuenta el pais al que pertenecen pueden ser descripciones de departamentos que se llaman igual en diferentes provincias, por lo cual la celda descripcion no identifica de forma univoca

In [ ]:
departamentos_id_unicos = df.groupby('departamento_destino_id')['departamento_destino'].nunique()
departamentos_id_unicos

,departamento_destino
departamento_destino_id,
2000,1
6007,1
6014,1
6021,1
6042,1
...,...
90105,1
90112,1
90119,1


Terminamos de confirmar que lo que identifica univocamente es el id de departamento ya que muestra una relacion 1-1

In [ ]:
# Obtenemos estadísticas básicas de las variables numéricas
df.describe()

,fecha,provincia_origen_id,departamento_origen_id,provincia_destino_id,departamento_destino_id,vaca,vaquillona,novillo,novillito,ternero,ternera,torito,toro,bueyes
count,168802,168802.000000,168802.000000,168802.000000,168802.000000,168802.000000,168802.000000,168802.000000,168802.000000,168802.000000,168802.000000,168802.000000,168802.000000,168802.000000
mean,2018-06-18 08:09:20.296679168,27.445125,27658.608666,29.831021,30064.051291,42.388337,35.813823,26.340950,37.402928,42.617516,33.294819,1.282023,3.313681,0.002927
min,2018-01-01 00:00:00,2.000000,2000.000000,2.000000,2000.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
25%,2018-04-01 00:00:00,6.000000,6511.000000,6.000000,6539.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
50%,2018-07-01 00:00:00,14.000000,14168.000000,14.000000,14119.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
75%,2018-09-01 00:00:00,42.000000,42056.000000,50.000000,50049.000000,32.000000,27.000000,8.000000,26.000000,33.000000,23.000000,0.000000,1.000000,0.000000
max,2018-12-01 00:00:00,94.000000,94015.000000,94.000000,94015.000000,10027.000000,8242.000000,5960.000000,8264.000000,6583.000000,6970.000000,1075.000000,1170.000000,262.000000
std,NaN,26.857114,26744.951837,29.547682,29416.051202,156.953718,132.454762,118.959993,147.214797,147.858749,125.992010,12.434034,15.230203,0.662959


Con el metodo describe() podemos ver que dentro el dataset refleja una distribucion de ganado demasiado extrema debido a que las medias identificadas son demasiado altas, las medianas estan bajas y hay mucha presencia de 0 lo que indica que la distribucion esta extremadamente distribuida y que encontraremos outliers grandes

In [ ]:
# Listamos los nombres de todas las columnas del dataset
df.columns

Index(['fecha', 'salido_de', 'ingresado_a', 'provincia_origen',
       'provincia_origen_id', 'departamento_origen', 'departamento_origen_id',
       'provincia_destino', 'provincia_destino_id', 'departamento_destino',
       'departamento_destino_id', 'vaca', 'vaquillona', 'novillo', 'novillito',
       'ternero', 'ternera', 'torito', 'toro', 'bueyes'],
      dtype='object')

###***Estructura del dataset***



El dataset contiene una gran cantidad de registros (más de 160.000 filas) y aproximadamente 20 columnas.

Cada fila representa un movimiento de ganado entre un origen y un destino específico.

Se pueden identificar distintos tipos de variables:

- Variables temporales: como la fecha
- Variables categóricas: provincias, departamentos, tipo de establecimiento
- Variables numéricas: cantidad de animales según categoría (vacas, novillos, terneros, etc.)

Esto permite analizar tanto la distribución geográfica como la composición del ganado en movimiento.


In [ ]:
# Contamos la cantidad de valores nulos por cada columna
df.isnull().sum()

,0
fecha,0
salido_de,0
ingresado_a,0
provincia_origen,0
provincia_origen_id,0
departamento_origen,0
departamento_origen_id,0
provincia_destino,0
provincia_destino_id,0
departamento_destino,0


Contabilizamos la cantidad de nulos por columna a fin de identificarlos

In [ ]:
# Porcentaje de valores nulos por columna
(df.isnull().sum() / len(df)) * 100

,0
fecha,0.0
salido_de,0.0
ingresado_a,0.0
provincia_origen,0.0
provincia_origen_id,0.0
departamento_origen,0.0
departamento_origen_id,0.0
provincia_destino,0.0
provincia_destino_id,0.0
departamento_destino,0.0


A partir del porcentaje terminamos de confirmar y asegurar que no poseemos ningun tipo de dato nulo por columna

###***Calidad de los datos***



Se realizó un análisis preliminar de calidad de los datos, observando que no se presentan valores nulos en ninguna de las variables.

Esto indica que el dataset se encuentra completo desde el punto de vista estructural, lo cual facilita su análisis.

Sin embargo, esto no descarta la posible existencia de otras inconsistencias, como errores en categorías o valores atípicos, que deberán analizarse en etapas posteriores.

####***A partir de los datos explorados del dataset logramos crear el siguiente diccionario***
| Nombre de la columna      | Descripción                                        | Tipo de dato | Valores nulos | Posible uso en análisis/modelado              |
| ------------------------- | -------------------------------------------------- | ------------ | ------------- | --------------------------------------------- |
| `fecha`                   | Fecha en que se realiza el movimiento              | `datetime64` | Ninguno       | Análisis temporal, estacionalidad, tendencias |
| `salido_de`               | Lugar desde el que sale el ganado                  | `object`     | Ninguno       | Análisis logístico y de trazabilidad          |
| `ingresado_a`             | Lugar al que llega el ganado                       | `object`     | Ninguno       | Identificación de destinos frecuentes         |
| `provincia_origen`        | Nombre de la provincia de origen                   | `object`     | Ninguno       | Distribución geográfica                       |
| `provincia_origen_id`     | ID de la provincia de origen                       | `int64`      | Ninguno       | Codificación para modelos                     |
| `departamento_origen`     | Nombre del departamento de origen                  | `object`     | Ninguno       | Segmentación regional                         |
| `departamento_origen_id`  | ID del departamento de origen                      | `int64`      | Ninguno       | Variables categóricas para ML                 |
| `provincia_destino`       | Nombre de la provincia de destino                  | `object`     | Ninguno       | Flujo interprovincial                         |
| `provincia_destino_id`    | ID de la provincia de destino                      | `int64`      | Ninguno       | Modelos predictivos y agrupaciones            |
| `departamento_destino`    | Nombre del departamento de destino                 | `object`     | Ninguno       | Análisis territorial                          |
| `departamento_destino_id` | ID del departamento de destino                     | `int64`      | Ninguno       | Relaciones origen-destino                     |
| `vaca`                    | Hembras que han parido al menos una vez            | `int64`      | Ninguno       | Producción reproductiva                       |
| `vaquillona`              | Hembras jóvenes que no han parido                  | `int64`      | Ninguno       | Proyección de stock reproductivo              |
| `novillo`                 | Machos castrados adultos                           | `int64`      | Ninguno       | Producción cárnica                            |
| `novillito`               | Machos castrados jóvenes                           | `int64`      | Ninguno       | Evolución de categorías bovinas               |
| `ternero`                 | Machos menores de un año                           | `int64`      | Ninguno       | Indicadores de nacimientos                    |
| `ternera`                 | Hembras menores de un año                          | `int64`      | Ninguno       | Crecimiento del rodeo                         |
| `torito`                  | Machos jóvenes mayores de un año sin servicio      | `int64`      | Ninguno       | Potencial reproductivo                        |
| `toro`                    | Machos destinados a reproducción                   | `int64`      | Ninguno       | Capacidad reproductiva                        |
| `bueyes`                  | Machos adultos castrados para trabajo o infértiles | `int64`      | Ninguno       | Uso productivo tradicional                    |





##3.1.C Justificación del dataset



El dataset seleccionado resulta adecuado para este trabajo porque permite analizar de forma concreta los movimientos de ganado bovino en Argentina, que es justamente el foco del problema planteado.

Se trata de un conjunto de datos amplio, con una gran cantidad de registros y variables, lo que posibilita realizar un análisis significativo y obtener conclusiones relevantes. Además, incluye tanto variables numéricas (cantidad de animales por categoría) como categóricas (provincias, departamentos, tipo de establecimiento), lo que permite abordar el análisis desde distintas perspectivas.

Otro aspecto importante es que incorpora una dimensión temporal, lo cual hace posible estudiar la evolución de los movimientos y detectar posibles patrones a lo largo del tiempo, algo directamente vinculado con las preguntas analíticas definidas.

Por otro lado, al tratarse de datos reales provenientes de un organismo oficial como SENASA, el dataset presenta ciertas imperfecciones (como valores faltantes o inconsistencias), lo cual lo hace especialmente útil para trabajar las etapas de limpieza y preparación de datos.

Finalmente, existe una relación directa entre los datos disponibles y los objetivos del trabajo, ya que el dataset permite analizar los flujos de ganado entre regiones, comparar comportamientos según categorías animales e identificar posibles patrones o concentraciones dentro del sistema.

En este sentido, el dataset no solo cumple con los requisitos técnicos de la consigna, sino que también resulta pertinente desde el punto de vista del problema planteado y las preguntas analíticas que guían el proyecto.

##4.1.A Preparacion de los datos

En el dataset no se encuentran datos faltantes, ni tampoco nulos

- Correccion de formatos y tipos de datos
- Limpieza de datos textuales si corresponde
- Cualquier otra transformacion necesaria (agregaciones, joins, creacion de nuevas variables)
Cada decisión de preparación debe estar justificada en el notebook


In [27]:
df.head(10)

,fecha,salido_de,ingresado_a,provincia_origen,provincia_origen_id,departamento_origen,departamento_origen_id,provincia_destino,provincia_destino_id,departamento_destino,departamento_destino_id,vaca,vaquillona,novillo,novillito,ternero,ternera,torito,toro,bueyes
0,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,25 DE MAYO,6854,212,150,0,342,931,566,0,21,0
1,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,9 DE JULIO,6588,35,0,0,0,96,104,0,4,0
2,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,ALBERTI,6021,0,0,0,0,31,9,0,0,0
3,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,BOLÍVAR,6105,0,0,0,0,29,39,0,0,0
4,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,BRAGADO,6112,0,0,0,0,11,14,0,0,0
5,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,CHIVILCOY,6224,0,0,0,0,35,30,0,0,0
6,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,GENERAL ALVEAR,6287,0,0,0,0,0,0,0,5,0
7,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,GENERAL VIAMONTE,6385,0,0,0,0,22,28,0,30,0
8,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,HIPÓLITO YRIGOYEN,6406,0,0,0,0,140,0,0,0,0
9,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,LOBOS,6483,50,7,0,6,10,10,0,0,0


In [26]:
columnas_upper = [
    "salido_de",
    "ingresado_a",
    "provincia_origen",
    "departamento_origen",
    "provincia_destino",
    "departamento_destino"
]

for col in columnas_upper:
    if col in df.columns:
        df[col] = df[col].astype(str).str.upper()
    else:
        print(f"No existe la columna: {col}")

df

,fecha,salido_de,ingresado_a,provincia_origen,provincia_origen_id,departamento_origen,departamento_origen_id,provincia_destino,provincia_destino_id,departamento_destino,departamento_destino_id,vaca,vaquillona,novillo,novillito,ternero,ternera,torito,toro,bueyes
0,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,25 DE MAYO,6854,212,150,0,342,931,566,0,21,0
1,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,9 DE JULIO,6588,35,0,0,0,96,104,0,4,0
2,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,ALBERTI,6021,0,0,0,0,31,9,0,0,0
3,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,BOLÍVAR,6105,0,0,0,0,29,39,0,0,0
4,2018-01-01,ESTABLECIMIENTO,ESTABLECIMIENTO,BUENOS AIRES,6,25 DE MAYO,6854,BUENOS AIRES,6,BRAGADO,6112,0,0,0,0,11,14,0,0,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
168797,2018-12-01,REMATE FERIA,FRIGORÍFICO,SANTIAGO DEL ESTERO,86,ROBLES,86161,SANTA FE,82,VERA,82133,0,37,0,0,0,0,26,0,0
168798,2018-12-01,REMATE FERIA,FRIGORÍFICO,SANTIAGO DEL ESTERO,86,ROBLES,86161,SANTIAGO DEL ESTERO,86,ROBLES,86161,92,176,19,1,0,0,2,12,0
168799,2018-12-01,REMATE FERIA,FRIGORÍFICO,SANTIAGO DEL ESTERO,86,ROBLES,86161,TUCUMÁN,90,CAPITAL,90084,0,7,0,13,0,0,0,0,0
168800,2018-12-01,REMATE FERIA,FRIGORÍFICO,SANTIAGO DEL ESTERO,86,ROBLES,86161,TUCUMÁN,90,LEALES,90056,25,20,0,12,0,0,8,0,0


##4.1.B Análisis descriptivo


- Calcular e interpreinterpretar medidas de tendencia central y dispersión para las variables
numéricas relevantes.
- Construir tablas de frecuencia y proporciones para las variables categóricas.
- Identificar y analizar valores atípicos: describir su impacto potencial en el análisis.

,fecha,salido_de,ingresado_a,provincia_origen,provincia_origen_id,departamento_origen,departamento_origen_id,provincia_destino,provincia_destino_id,departamento_destino,departamento_destino_id,vaca,vaquillona,novillo,novillito,ternero,ternera,torito,toro,bueyes
0,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,25 de Mayo,6854,212,150,0,342,931,566,0,21,0
1,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,9 de Julio,6588,35,0,0,0,96,104,0,4,0
2,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Alberti,6021,0,0,0,0,31,9,0,0,0
3,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Bolívar,6105,0,0,0,0,29,39,0,0,0
4,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Bragado,6112,0,0,0,0,11,14,0,0,0
5,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Chivilcoy,6224,0,0,0,0,35,30,0,0,0
6,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,General Alvear,6287,0,0,0,0,0,0,0,5,0
7,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,General Viamonte,6385,0,0,0,0,22,28,0,30,0
8,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Hipólito Yrigoyen,6406,0,0,0,0,140,0,0,0,0
9,2018-01-01,Establecimiento,Establecimiento,Buenos Aires,6,25 de Mayo,6854,Buenos Aires,6,Lobos,6483,50,7,0,6,10,10,0,0,0


,departamento_destino
0,25 DE MAYO
1,9 DE JULIO
2,ALBERTI
3,BOLÍVAR
4,BRAGADO
...,...
168797,VERA
168798,ROBLES
168799,CAPITAL
168800,LEALES


##4.1.C Análisis de relaciones entre variables

- Explorar correlaciones y asociaciones entre variables.
- Conectar los hallazgos con las preguntas analíticas formuladas en la Entrega 1.

##4.1.D Visualizaciones

- Producir al menos 5 (cinco) visualizaciones distintas que aporten información
sustantiva al análisis.
- Cada visualización debe incluir título, etiquetas claras en los ejes y una interpretación
escrita en celda Markdown.
- Aplicar principios de diseño: elegir el tipo de gráfico adecuado para cada variable o
relación, evitar gráficos decorativos o redundantes.

##4.1.E Sintesis del EDA

Escribir un apartado de síntesis que reponda:
- ¿Cuáles son los principales hallazgos del análisis exploratorio?
- ¿Qué patrones o tendencias se observan?
- ¿Qué preguntas de la Entrega 1 pueden responderse ya con el EDA, y cuales
requieren un abordaje más avanzado?
- ¿Hay indicios de que un modelo predictivo podría ser útil? ¿Por qué?

##7. Conclusión



En esta primera etapa del trabajo se logró definir el problema, entender el contexto del dominio y realizar una exploración inicial del dataset.

Se identificaron las principales características de los datos y algunos aspectos a mejorar en términos de calidad.

Este análisis inicial sirve como base para las siguientes etapas, donde se profundizará en el análisis exploratorio y se buscarán patrones más concretos en los datos.